### PR TITLE
GP2 1068: black and isort

### PR DESCRIPTION
**Pull request by stevejalim**
GP2-1068: Set up black + isort + support use of pre-commit

* add Black and related deps to test requirements
* add minimal Black config in pyproject.toml which will be used by default
* config for isort so that it doesn't conflict with Black
* add makefile commands to use black in check or write mode (`make checks` or `make autoformat`)
* update CI config to run `black --check` in CI, too
* add .pre-commit-config.yaml for pre-commit (see pre-commit.com)
* DISABLE pep8 checks in codeclimate - we already have flake8 and now we have black running in github CI too

* UPDATE VERSION to 22.9.1

* Apply Black and isort to the entire codebase - this brings it in to line with Black, while also ensures that if you use pre-commit (which also runs isort) we starting from a clean slate.

### CODE REVIEW

**Comment by stevejalim** [comment]
@depsiatwal @webbyfox This changeset is only black/isort and no new features - so I've made it a patch release - do you agree? If so, will all users of this client need manually upgrading or do they have `maj.min.*` set in their `requirements.in`?

**Comment by stevejalim** [comment]
Actually I just noticed that great-cms is on `22.4.*`... I don't want to break things pre-Christmas, so is that a safe bump?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Changelog
 
+## [22.9.1](https://pypi.org/project/directory-api-client/22.9.0/) (2020-12-08)
```
Comment (webbyfox) [inline CHANGELOG.md:3]: URL pointing to `2.9.0`

**Review by webbyfox — APPROVED** [review]
LGTM, please update url in CL and good to merge

**Comment by stevejalim** [comment]
@webbyfox updated - thanks for the laser eyes!

**Pull request merged**